app/api/student.py

```diff
@@ -1,40 +1,50 @@
+import logging
+import typing as tp
+from db.db import Database, BriefQueue, Queue, ComparatorType
 
 
 class Student:
 
-    def __init__(self, student_tg_name):
-        self.student_tg_name = student_tg_name
-        self.groups = []
-        self.review_queues = []
+    def __init__(self, student_tg_id: int, database: Database) -> None:
+        self.student_tg_id = student_tg_id
+        self.database = database
+        self.is_subscribed_review_queue_enroll_start_notifications = True
 
 
-    def join_group(self, group_name, invite_code):
-        pass
+    def join_group(self, invite_code: str) -> None:
```

Comment:
Looks like UI should be aware of result of joining to be able to answer the user if something goes wrong, but currently there is not exception nor status return from here

Reply:
I agree about the presence of exceptions, there really aren’t enough of them

Reply:
Added https://github.com/six-nine/edu_queue/pull/1/commits/3fd1f243122f97c5129c50350b8bbc95d6dcdb40 'StudentJoinGroupException' and 'StudentLeaveGroupException'